### ISSUE TITLE
Custom Dual Wielding and Forge Controls don't Overlap Existing Controls.

### ISSUE THREAD

**Issue by theweion** [ElDewrito/ElDorito#138]
It is currently only programmed to overlap the default controls, but if the player uses custom controls (Halo CE Style for example) then the dual wield/forge controls become impossible to map.

**Comment by Shockfire** [comment]
Can you explain this further please? What exactly are you doing, what do you expect to happen, and what happens instead?

**Comment by theweion** [comment]
For example, I have changed the Zoom, Throw Grenade, and Melee commands to: Middle Mouse Button, Right Mouse Button, and F respectively. This causes the Dual Wield Shoot Left control to unbind because it was already assigned to Throw Grenade.

I have been able to "fix" this by setting my controls in Visual Studio before compiling it, the only solutions I can think is either figure out a way of bypassing the check for overlapping keys, or have an external config file for the controls that are automatically run at launch.

**Comment by NoShotz** [comment]
You can change the binds for the secondary weapon shoot in the settings
menu...

**Comment by NoShotz** [comment]
Also tag updates for the keybinding system will allow overlapping keys i
believe, but they aren't put

**Comment by NoShotz** [comment]
Public*

**Comment by theweion** [comment]
I know you can change them, the issue is that they don't overlap like so:

https://my.mixtape.moe/peylga.mp4


> Also tag updates for the keybinding system will allow overlapping keys i
> believe, but they aren't put public.

Oh that is good.

I also found this these two snippets of code under `ModuleInput.cpp` and I have been testing out some changes:

```
bindings.PrimaryKeys[eGameActionReloadLeft] = eKeyCodeZ; // TODO: Find a better key?
bindings.PrimaryKeys[eGameActionPickUpLeft] = eKeyCodeZ; // TODO: Find a better key?
```

Binding the `eGameActionReloadLeft` and `eGameActionPickUpLeft` to `eKeyCodeE` allows the player to easily reload the left weapon and pickup other items, also Tapping and Holding E can be used to change the Right and Left weapons respectively.

**Comment by theTwist84** [comment]
If you look in dewrito_prefs.cfg you can overlap them there.

**Comment by NoShotz** [comment]
E is th pickup key, so that wont be the default

**Comment by NoShotz** [comment]
Also we cant do the tap and hold stuff

**Comment by theweion** [comment]
> If you look in dewrito_prefs.cfg you can overlap them there.

Oh that is great, I didn't notice that it got changed.

> If you look in dewrito_prefs.cfg you can overlap them there.
> Also we cant do the tap and hold stuff

https://my.mixtape.moe/uanowq.mp4

This video illustrates using E to Pickup weapons/dual wield, reloading the other weapon and entering vehicles. At the end it shows the weapons swapping between hands depending on if I hold or press the E key.

**Comment by NoShotz** [comment]
But when keybinding, there is no way to differentiate between holding or
pressing

**Comment by Shockfire** [comment]
From the commit message for df93c123049df52689e15d8c09b07263f6b88cac:
>Also, we will be editing the SSL scripts included with 0.5.1.0 in order to allow keys to be bound to multiple actions (needed for some of the dual wield stuff).

Dual wield pickup isn't bound to E by default because if there are two of the same weapon on the ground next to each other, then it makes you pick up both of them unless you tap the key really lightly. This isn't always what the player wants.

**Comment by theweion** [comment]
Strange, in my tests I can literally just press the E key normally over two weapons of the same type and pick up one. I only pick up both of them if I hold down the key for a second.

I am using Cherry MX Blues so it is literally impossible for me to lightly tap any key.
